typescript/packages/plugins/dragonswap/src/dragonswap.service.ts

```diff
@@ -0,0 +1,165 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { ERC20_ABI } from "./abi/erc20abi";
+import { factoryAbi } from "./abi/factoryabi";
+import { quoterv2abi } from "./abi/quoterv2abi";
+import { swaprouterabi } from "./abi/swaprouterabi";
+import { POSITION_MANAGER_ABI } from "./abi/nftpositionmanager";
+import { DragonswapConfig } from "./dragonswap.plugin";
+import {
+    AddLiquidityParameters,
+    CollectFeesParameters,
+    ExactInputSingleParameters,
+    RemoveLiquidityParameters,
+} from "./parameters";
+
+export class DragonswapService {
+    constructor(private readonly config: DragonswapConfig) {}
+
+    @Tool({
+        name: "swap_exact_input_single_on_dragonswap",
+        description: "Swap tokens on Dragonswap using exact input amount",
+    })
+    async exactInputSingle(walletClient: EVMWalletClient, parameters: ExactInputSingleParameters) {
+        // Approve token if needed
+        await walletClient.sendTransaction({
+            to: parameters.tokenIn as `0x${string}`,
+            abi: ERC20_ABI,
+            functionName: "approve",
+            args: [this.config.routerAddress, parameters.amountIn],
+        });
+
+        const connectedaddress = await walletClient.getAddress();
+
+        const tx = await walletClient.sendTransaction({
+            to: this.config.routerAddress as `0x${string}`,
+            abi: swaprouterabi,
+            functionName: "exactInputSingle",
+            args: [{
+                tokenIn: parameters.tokenIn,
+                tokenOut: parameters.tokenOut,
+                fee: parameters.fee,
+                recipient: parameters.recipient || connectedaddress,
+                deadline: parameters.deadline,
+                amountIn: parameters.amountIn,
+                amountOutMinimum: parameters.amountOutMinimum,
+                sqrtPriceLimitX96: parameters.sqrtPriceLimitX96 || 0,
+            }],
+        });
+
+        return tx.hash;
+    }
+
+    @Tool({
+        name: "get_quote_from_dragonswap",
+        description: "Get quote for a swap on Dragonswap",
+    })
+    async getQuote(walletClient: EVMWalletClient, parameters: ExactInputSingleParameters) {
+        
+        const quote = await walletClient.sendTransaction({
```

Comment:
Is this a sendTransaction? I can't check the contract since I can't find it on Sei explorer.